packages/block-editor/src/components/global-styles/image-settings-panel.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	ToggleControl,
+} from '@wordpress/components';
+import { __, _x } from '@wordpress/i18n';
+
+export function useHasImageSettingsPanel( name, settings, userSettings ) {
+	// Note: If lightbox userSettings exists, that means
+	// they were defined via the Global Styles UI and
+	// will NOT be a boolean value or contain the `allowEditing`
+	// property, so we should show the settings panel in those cases.
+	return (
+		( name === 'core/image' && settings?.lightbox?.allowEditing ) ||
+		!! userSettings?.lightbox
+	);
+}
+
+export default function ImageSettingsPanel( {
+	onChange,
+	userSettings,
+	settings,
```

Comment:
Should these props be named `value` and `inheritedValue` instead of `userSettings` and `settings`. I understand that it's a bit different than the other components like `TypographyPanel` because this one updates the settings rather than the styles but there's a small precedent. The `DimentionsPanel` also updates the "layout" which is saved in "settings". 

I like to think of the components in separation (regardless of where the value ends up saved). If a component has `onChange` ideally, it also has `value`. `settings` is used in general for things that configure the behavior of the component, not the thing that is changed by the component.

Reply:
> If a component has onChange ideally, it also has value.

I gotta acknowledge that it's a convention in React, but at the same time, having a `value` prop in this situation erases the meaning of the original prop (passed from the parent). 

For example:

https://github.com/WordPress/gutenberg/blob/596966838a8edff52816a981e65e9ad4ddbf1ab6/packages/block-editor/src/components/global-styles/image-settings-panel.js#L40-L42

When I look at this, I know it's dealing with the `lightbox` **setting**. If `settings` were renamed to `inheritedValue` I wouldn't know if it's a setting, a style, or something else. At least this is something that confused me when I looked at other panels initially 🙂

Reply:
I understand that it might confuse you but I think consistency is better than personal opinion in this case. I value your opinion but if we adopt something I think we should adopt in all components and not just one. Also this has been discussed previously here https://github.com/WordPress/gutenberg/discussions/37064

Reply:
I see! I don't feel very strongly about this, and indeed whether it's confusing or not is subjective/personal 🙂. I'll update it in a follow-up PR. Thanks Riad!

Reply:
Done in https://github.com/WordPress/gutenberg/pull/54593

Reply:
I am testing image light box with WP 6.4-beta1 without Gutenberg plugin. The image light box doesn't work when I create gallery but it works with normal image block. 
![Screenshot 2023-09-28 at 10 11 17 AM](https://github.com/WordPress/gutenberg/assets/24829573/47fd050a-0695-4d03-a0ab-eb8b49dbafa3)

Reply:
@iwangdu Thanks for taking a look! I just tested in the WP 6.4-beta1 and it's working for me. Note that we don't have full gallery support, so you won't be able to toggle between the images while the lightbox is open, though the zooming should work as expected.

What happens when you click on a gallery image on the front end? Does the image expand?